### DISCUSSION TITLE
Let visual selection synchronize with vscode selection

### DISCUSSION THREAD

**Discussion by shaolun-huang** [vscode-neovim/vscode-neovim#1342] [Q&A]
Is there a way to set visual selection synchronizing with vscode selection? That is, if I select multiple lines in the visual mode, how to let vscode recognize the selection?

**Comment by theol0403** [comment]
Yes, it is in the next release

**Reply by shellwhale** [reply]
Hello, is that out? How does that work?

When I make a selection in visual mode and that I press `i` it doesn't go to insert mode

**Reply by theol0403** [reply]
Why do you expect it to do that??

**Reply by xiyaowong** [reply]
The functionality of vscode-neovim is the intersection of vscode and neovim.

**Reply by xiyaowong** [reply]
Try https://github.com/vscode-neovim/vscode-multi-cursor.nvim#start-editing
```lua
local k = vim.keymap.set
k({ 'x' }, 'I', M.start_left_edge)
k({ 'x' }, 'A', M.start_right)
```

**Reply by shellwhale** [reply]
I'm sorry @xiyaowong I'm fairly new to this, what's `M` ?
`E5113: Error while calling lua chunk: C:\Users\shellwhale\AppData\Local\nvim\init.lua:22: attempt to index global 'M' (a nil value)`

I can't find a declaration for it in your readme file

**Reply by xiyaowong** [reply]
local M = require('vscode-multi-cursor')

**Reply by shellwhale** [reply]
Works like a charm! Many thanks for your work guys
